common/djangoapps/credo_modules/middleware.py

```diff
@@ -33,3 +35,24 @@ def process_request(self, request):
                     redirect_url = ''.join([redirect_url, '?next=', next_page])
                     return HttpResponseRedirect(redirect_url)
         return None
+
+    @staticmethod
+    def is_graded_block(request):
+        def get_block_id():
+            def grep_block_id(s):
+                return re.search('block-v1:.*?@.*?@[a-z0-9]*', s)
```

Comment:
you need not to reimplement this regular expression. Take a look here:
https://github.com/CredoReference/edx-platform/blob/master/lms/envs/common.py#L578
you may use `USAGE_KEY_PATTERN`